src/welcome-bare-metal.md

```diff
@@ -0,0 +1,32 @@
+# Welcome to Bare Metal Rust
+
+This is a standalone one-day course about bare-metal Rust, aimed at people who are familiar with the
+basics of Rust (perhaps from completing the Comprehensive Rust course), and ideally also have some
+experience with bare-metal programming in some other language such as C.
+
+Today we will talk about 'bare-metal' Rust: running Rust code without an OS underneath us. This will
+be divided into several parts:
+
+ * What is `no_std` Rust?
+ * Writing firmware for microcontrollers.
+ * Writing bootloader / kernel code for application processors.
+ * Some useful crates for bare-metal Rust development.
+
+For the microcontroller part of the course we will use the [BBC micro:bit](https://microbit.org/) v2
+as an example. It's a [development board](https://tech.microbit.org/hardware/) based on the Nordic
+nRF51822 microcontroller with some LEDs and buttons, an I2C-connected accelerometer and compass, and
+an on-board SWD debugger.
+
+To get started, install some tools we'll need later:
+
+```bash
+sudo apt install gdb-multiarch picocom
```

Comment:
I needed `libudev-dev` as well:

```suggestion
sudo apt install gdb-multiarch picocom libudev-dev
```

Reply:
That's weird, what happens if you don't install it?

Reply:
I cannot install `cargo-embed` without it. It fails with

```
error: failed to run custom build command for `hidapi v2.2.0`

Caused by:
  process didn't exit successfully: `/tmp/cargo-installegDUk8/release/build/hidapi-1b03c572ab5df1f8/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=LIBUDEV_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=LIBUDEV_STATIC
  cargo:rerun-if-env-changed=LIBUDEV_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

  --- stderr
  thread 'main' panicked at 'Unable to find libudev: `"pkg-config" "--libs" "--cflags" "libudev"` did not exit successfully: exit status: 1
  error: could not find system library 'libudev' required by the 'hidapi' crate

  --- stderr
  Package libudev was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libudev.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'libudev' found
  ', /usr/local/google/home/mgeisler/.cargo/registry/src/github.com-1ecc6299db9ec823/hidapi-2.2.0/build.rs:57:54
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: failed to compile `cargo-embed v0.16.0`, intermediate artifacts can be found at `/tmp/cargo-installegDUk8`
```

Reply:
Aha! That makes sense then, added.